### PR TITLE
Standardize all environment variable boolean configuration in python's setup.py

### LINKED ISSUE #24498: Reading of options from env variables is broken in setup.py (e.g. BUILD_WITH_BORING_SSL_ASM)
### What version of gRPC and what language are you using?
1.32.0

### What operating system (Linux, Windows,...) and version?
Linux

### What runtime / compiler are you using (e.g. python version or version of gcc)
GCC 10

### What did you do?
Noticed that there's no possible value that can be passed through as BUILD_WITH_BORING_SSL_ASM that will disable boringssl https://github.com/grpc/grpc/blob/master/setup.py#L109-L110

Since there's no int() conversion of input (or similar), all possible inputs will be regarded as true. 

### What did you expect to see?
It's possible to disable boringssl asm when using system ssl

### What did you see instead?
It's not possible to disable boringssl asm when using system ssl

### PR DESCRIPTION
**Pull request by emkornfield**
@jtattermusch @lidizheng

Fixes #24498

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -300,7 +300,8 @@ def check_linker_need_libatomic():
 asm_files = []
 
 asm_key = ''
-if BUILD_WITH_BORING_SSL_ASM and not BUILD_WITH_SYSTEM_OPENSSL:
+if (BUILD_WITH_BORING_SSL_ASM.upper() == 'TRUE'
+    or BUILD_WITH_BORING_SSL_ASM == '1') and not BUILD_WITH_SYSTEM_OPENSSL:
```
Comment (lidizheng) [inline setup.py:304]: nit: Technically, users can set this flag to any value and they should be evaluated into True, except `False, 0, ''`. How about we detect the falsify values intead?

**Review by lidizheng — COMMENTED** [review]
Thanks for putting up this PR!

File: setup.py (hunk 1 above)
Comment (emkornfield) [reply setup.py]: done.

**Review by lidizheng — APPROVED** [review]
LGTM.

File: setup.py [hunk 2]
```diff
@@ -300,7 +300,8 @@ def check_linker_need_libatomic():
 asm_files = []
 
 asm_key = ''
-if BUILD_WITH_BORING_SSL_ASM and not BUILD_WITH_SYSTEM_OPENSSL:
+if (BUILD_WITH_BORING_SSL_ASM.upper() not in ['FALSE', '0', ''] and
```
Comment (jtattermusch) [inline setup.py:303]: Fair enough, but as pointed out in https://github.com/grpc/grpc/issues/24498#issuecomment-778123526 there are plenty of other options that can be set for setup.py  and by changing the behavior for BUILD_WITH_BORING_SSL_ASM only, we're basically introducing an inconsistency (and having some setup.py with one behavior and some with a different behavior is something we really don't want).

My suggestion is to introduce a private function e.g. `_get_build_option_bool_value(env_name))` that has the `upper() not in ['FALSE', '0', '']` logic in it and then use it for all the boolean build options in setup.py - there are quite a few.

**Review by jtattermusch — CHANGES_REQUESTED** [review]
Thanks for the contribution but I think we really need to make all the boolean build options behave consistently, see comment

**Comment by jtattermusch** [comment]
Also not that the sanity check is failing (yapf code formatting):
https://source.cloud.google.com/results/invocations/2c0a0875-171c-4b32-bba1-35b38964b206/targets/github%2Fgrpc%2Frun_tests%2Fsanity_linux_dbg_native%2Ftools%2Fdistrib%2Fyapf_code.sh/tests

**Comment by emkornfield** [comment]
@jtattermusch I tried to address you comments.  Would you mind taking another look?  Also how can I run lint/formatting locally (I can't seem to find the instructions, which I feel like I had at one point).

**Review by jtattermusch — APPROVED** [review]
LGTM once the tests pass.

**Comment by emkornfield** [comment]
Needed to rebase.

**Comment by jtattermusch** [comment]
The interop test failure is https://github.com/grpc/grpc/issues/25652.

**Pull request merged**